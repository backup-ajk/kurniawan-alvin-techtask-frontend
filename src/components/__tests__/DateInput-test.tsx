import React from 'react';
import renderer from 'react-test-renderer';

import { DateInput } from '../';
import ReactDatePicker from 'react-datepicker';

it('should render Date Input successfully', () => {
  let testDate = new Date();
  const onChangeDate = (newDate: Date) => {
    testDate = newDate;
  };
  const input = renderer
    .create(<DateInput value={testDate} onChangeDate={onChangeDate} />)
    .toJSON();
  expect(input).toMatchSnapshot();
});

it('should trigger on change date function date is selected', () => {
  let testDate = new Date();
  const tomorrow = new Date(testDate);
  tomorrow.setDate(tomorrow.getDate() + 1);

  const mockChangeDate = (newDate: Date) => {
    testDate = new Date(newDate);
  };

  const rendered = renderer.create(
    <DateInput value={testDate} onChangeDate={mockChangeDate} />,
  );
  const input = rendered.root.findByType(ReactDatePicker);
  input.props.onChange(tomorrow);

  expect(testDate).toEqual(tomorrow);
});
