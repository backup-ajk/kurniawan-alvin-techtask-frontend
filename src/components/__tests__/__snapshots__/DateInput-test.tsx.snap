// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Date Input successfully 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderColor": "#0064D2",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 2,
          "cursor": "pointer",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10,
        }
      }
    >
      <span
        style={
          Object {
            "color": "#212121",
            "fontSize": 18,
          }
        }
      >
        21-03-2020
      </span>
    </div>
  </div>
</div>
`;
